### PR TITLE
Remove duplicate `yarn.lock`, configure `yarn exe` for scripts

### PR DESCRIPTION
**Pull request by kachkaev**
This PR helps unblock #169 

I’ve noticed us having disjoint dependency versions in `.github/workflows/algolia/package.json` and other `package.json` files. The former file is not reached by `check-dependency-version-consistency` (#165) because `.github/workflows/algolia` is not a workspace. Thus, when we upgrade dependencies, we have to either ignore `.github/workflows/algolia/package.json` or update both `yarn.lock` files. This is not sustainable.

This PR makes sure we have only one `yarn.lock` file in the whole project. To run the Algolia script, I created `yarn exe` for convenience. See https://github.com/blockprotocol/blockprotocol/pull/157#discussion_r795958433 for context.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,10 +34,14 @@
     ]
   },
   "devDependencies": {
+    "@swc/core": "^1.2.135",
+    "@swc/helpers": "^0.3.2",
```
Comment (kachkaev) [inline package.json:38]: Enables fast transpilation in `ts-node`
https://github.com/TypeStrong/ts-node#bundled-swc-integration

File: package.json [hunk 2]
```diff
@@ -33,10 +34,14 @@
...
     "@typescript-eslint/eslint-plugin": "4.29.0",
     "@typescript-eslint/parser": "4.29.0",
+    "algoliasearch": "^4.12.0",
```
Comment (kachkaev) [inline package.json:41]: Moved from `.github/workflows/algolia`, which was not a workspace

File: package.json [hunk 3]
```diff
@@ -33,10 +34,14 @@
...
     "check-dependency-version-consistency": "^1.6.0",
     "concurrently": "^6.5.1",
+    "dotenv-flow": "3.2.0",
```
Comment (kachkaev) [inline package.json:44]: Helps us load `.env`, `.env.local` etc. automatically. See https://github.com/blockprotocol/blockprotocol/pull/157#discussion_r795958433

File: packages/create-block-app/create-block-app.js [hunk 4]
```diff
@@ -20,7 +20,8 @@ const templatePackageName = "block-template";
     process.exit();
   }
 
-  const blockPath = process.argv[3] ?? blockName;
+  // @todo: replace with `process.argv[3] ?? blockName` after upgrading ESLint
+  const blockPath = process.argv[3] ? process.argv[3] : blockName;
```
Comment (kachkaev) [inline packages/create-block-app/create-block-app.js:24]: Getting syntax error when running `yarn lint:eslint`. Might be to do with outdated built-in babel. I’ll update ESLint to fix that, which might be handy before #168 anyway:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.4.0

YOUR TYPESCRIPT VERSION: 4.5.4

Please only submit bug reports when using the officially supported version.

=============
```

Comment (kachkaev) [reply packages/create-block-app/create-block-app.js:24]: Same problem with `} catch {` below

File: scripts/sync-algolia-index.ts [hunk 5]
```diff
@@ -111,8 +115,8 @@ const generateAlgoliaRecords: () => AlgoliaRecord[] = () => {
 
 const syncAlgoliaIndex = async () => {
   const client = algoliasearch(
-    process.env.ALGOLIA_PROJECT,
-    process.env.AGOLIA_WRITE_KEY,
+    process.env.ALGOLIA_PROJECT ?? "",
+    process.env.AGOLIA_WRITE_KEY ?? "",
```
Comment (kachkaev) [inline scripts/sync-algolia-index.ts:119]: Typescript finally checks this file. It complained about assigning `undefined` to `string`. Might be worth adding `envalid` and checking vars there for a better error message. I can do that in a follow-up.

File: scripts/sync-algolia-index.ts [hunk 6]
```diff
@@ -149,14 +153,18 @@ const syncAlgoliaIndex = async () => {
...
+      `Error while running Algolia Update: ${
+        // @ts-expect-error -- TS is a bit too strict here. We can‘t use instanceof Error because Algolia errors are plain objects
+        error?.message ?? error
+      }`,
+    );
   }
 };
 
-void main();
+void script();
```
Comment (kachkaev) [inline scripts/sync-algolia-index.ts:170]: Justification: https://github.com/blockprotocol/blockprotocol/pull/157/files#r795996557

File: package.json (hunk 1 above)
Comment (thehabbos007) [reply package.json:38]: SWC :rocket:

**Comment by thehabbos007** [comment]
Just out of curiosity, without supplying a path to a script with `yarn exe` is that environment a REPL that is usable to do ad-hoc testing?

**Comment by kachkaev** [comment]
Hmmm never thought about REPL in this context, but yeah, it looks so! `yarn exe` is essentially a wrapped version of `node`. We currently run `ts-node` in it, but it will become something like `node --require=dotenv-flow/config --loader ts-node/esm/transpile-only --experimental-specifier-resolution=node` in the future when we unlock native ESM in scripts (https://github.com/TypeStrong/ts-node/issues/1007).

**Review by thehabbos007 — APPROVED** [review]
Looks good to me and works locally :)

**Pull request merged**